### PR TITLE
std.c: update netbsd/openbsd mman constants

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
- Definitions for [NetBSD ](https://github.com/NetBSD/src/blob/76ebf67f83026cc338ca5a06296c302c718ae8b6/sys/sys/mman.h#L82)and [OpenBSD](https://github.com/openbsd/src/blob/edc829431effa915c16b067b5da3c6ab86b7fbca/sys/sys/mman.h#L102).
- Note that PROT_MPROTECT while not necessarily vital allows to do it programmatically w/o using paxctl command line.

File: lib/std/c/netbsd.zig [hunk 1]
```diff
@@ -579,6 +579,12 @@ pub const PROT = struct {
     pub const READ = 1;
     pub const WRITE = 2;
     pub const EXEC = 4;
+    pub fn PROTECT(flag: u32) u32 {
+        return flag << 3;
+    }
+    pub fn PROTECT_EXTRACT(flag: u32) u32 {
+        return (flag << 3) & 0x7;
```
Comment (ifreund) [inline lib/std/c/netbsd.zig:586]: the C code from the header you linked has a right shift, not a left shift:
```c
#define        PROT_MPROTECT_EXTRACT(x)        (((x) >> 3) & 0x7)
```

**Review by ifreund — CHANGES_REQUESTED** [review]

File: lib/std/c/netbsd.zig (hunk 1 above)
Comment (devnexen) [reply lib/std/c/netbsd.zig:586]: true, thx.

File: lib/std/c/netbsd.zig [hunk 2]
```diff
@@ -579,6 +579,12 @@ pub const PROT = struct {
     pub const READ = 1;
     pub const WRITE = 2;
     pub const EXEC = 4;
+    pub fn MPROTECT(flag: u32) u32 {
+        return flag >> 3;
```
Comment (ifreund) [inline lib/std/c/netbsd.zig:583]: Now this shift does not match the header.

**Review by ifreund — CHANGES_REQUESTED** [review]

**Review by ifreund — APPROVED** [review]
LGTM, thanks!

**Pull request merged**